WalletWasabi.Fluent/Controls/TagsBox.xaml.cs

```diff
@@ -0,0 +1,325 @@
+            // know that the focus changed sometime in the past.  To determine
+            // if you currently have the focus you need to do consult the
+            // FocusManager (see HasFocus()).
+
+            var wasFocused = _isFocused;
+            _isFocused = hasFocus;
+
+            if (hasFocus)
+            {
+                if (!wasFocused)
+                {
+                    _autoCompleteBox?.Focus();
+                }
+            }
+
+            PseudoClasses.Set(":focus", hasFocus);
+            _isFocused = hasFocus;
+        }
+
+        private void OnDropDownClosed(object? sender, EventArgs e)
+        {
+            if (_autoCompleteBox is null)
+            {
+                return;
+            }
+
+            var currentText = _autoCompleteBox.Text ?? "";
+
+            if (currentText.Length == 0 ||
+                !(_autoCompleteBox.SelectedItem is string selItem) ||
+                selItem.Length == 0 ||
+                currentText != selItem)
+            {
+                return;
+            }
+
+            AddTag(currentText.Trim());
+
+            BackspaceLogicClear();
+
+            _autoCompleteBox.ClearValue(AutoCompleteBox.SelectedItemProperty);
+            Dispatcher.UIThread.Post(() => { _autoCompleteBox.ClearValue(AutoCompleteBox.TextProperty); });
+        }
+
+        private void BackspaceLogicClear()
+        {
+            _backspaceEmptyField2 = _backspaceEmptyField1 = false;
+        }
+
+        private void OnTextChanged(object? sender, EventArgs e)
+        {
+            if (_autoCompleteBox is null)
+            {
+                return;
+            }
+
+            var currentText = _autoCompleteBox.Text ?? "";
+            var currentTextTrimmed = currentText.Trim();
+
+            if (RestrictInputToSuggestions && Suggestions is IList<string> suggestions)
+            {
+                var keywordIsInSuggestions = suggestions.Any(x => x.Equals(currentTextTrimmed,
+                    StringComparison.InvariantCultureIgnoreCase));
+
+                if (!keywordIsInSuggestions)
+                {
+                    return;
+                }
+            }
+
+            if (!_isInputEnabled ||
+                currentText.Length < 1 ||
+                string.IsNullOrEmpty(currentTextTrimmed) ||
+                !currentText.EndsWith(' '))
+            {
+                return;
+            }
+
+            AddTag(currentTextTrimmed);
+
+            BackspaceLogicClear();
+
+            Dispatcher.UIThread.Post(() => { _autoCompleteBox.ClearValue(AutoCompleteBox.TextProperty); });
+        }
+
+        private void OnKeyUp(object? sender, KeyEventArgs e)
+        {
+            if (_autoCompleteBox is null) return;
+
+            var str = _autoCompleteBox?.Text ?? "";
+
+            _backspaceEmptyField2 = _backspaceEmptyField1;
+            _backspaceEmptyField1 = str.Length == 0;
+
+            var strTrimmed = str.Trim();
```

Comment:
why not Trim str from the beggining...

prefer to name variables without abbreviations... `keyword`, `search text` or any other word that describes what it really is.